packages/lit-analyzer/src/cli/analyze-command.ts

```diff
@@ -111,7 +111,11 @@ export async function analyzeCommand(globs: string[], config: LitAnalyzerCliConf
 		printText(statsText, config);
 	}
 
-	return stats.diagnostics <= (config.maxWarnings || 0) || config.maxWarnings === -1;
+	if (stats.errors !== 0) {
+		return false;
+	}
+
+	return stats.warnings <= (config.maxWarnings || 0) || config.maxWarnings === -1;
```

Comment:
```suggestion
	return stats.warnings <= config.maxWarnings! || config.maxWarnings === -1;
```

@runem I don't know if you prefer the `|| 0` for some reason.
If not, you can accept this proposal. :)

Reply:
@43081j got it, thanks!